lang/cs/Org.Apache.REEF.Network/Naming/NameClient.cs

```diff
@@ -64,13 +65,33 @@ public class NameClient : INameClient
         /// <param name="remoteAddress">The ip address of the NameServer</param>
         /// <param name="remotePort">The port of the NameServer</param>
         [Inject]
-        public NameClient(
+        private NameClient(
             [Parameter(typeof(NamingConfigurationOptions.NameServerAddress))] string remoteAddress, 
             [Parameter(typeof(NamingConfigurationOptions.NameServerPort))] int remotePort)
         {
             IPEndPoint remoteEndpoint = new IPEndPoint(IPAddress.Parse(remoteAddress), remotePort);
             Initialize(remoteEndpoint);
             _disposed = false;
+            _cache = TangFactory.GetTang().NewInjector().GetInstance<NameCache>();
```

Comment:
Provided two constructors to maintain backward compatibility....Should I
remove the previous one.

On Mon, Jul 27, 2015 at 7:30 AM, Markus Weimer <notifications@github.com>
wrote:

> In lang/cs/Org.Apache.REEF.Network/Naming/NameClient.cs
> <https://github.com/apache/incubator-reef/pull/318#discussion_r35541407>:
>
> >              [Parameter(typeof(NamingConfigurationOptions.NameServerAddress))] string remoteAddress,
> >              [Parameter(typeof(NamingConfigurationOptions.NameServerPort))] int remotePort)
> >          {
> >              IPEndPoint remoteEndpoint = new IPEndPoint(IPAddress.Parse(remoteAddress), remotePort);
> >              Initialize(remoteEndpoint);
> >              _disposed = false;
> > +            _cache = TangFactory.GetTang().NewInjector().GetInstance<NameCache>();
>
> why use Tang here? Can't the cache be injected?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-reef/pull/318/files#r35541407>.
>